docs/i18n/30-formatting.md

```diff
@@ -0,0 +1,297 @@
+
+For [ICU message formatting](#icu-message-formatting):
+
+-   `supplemental/likelySubtags`
+-   `supplemental/plurals`
+
+For [date/time formatting](#date-and-number-formatting):
+
+-   `main/{locale}/ca-gregorian`
+-   `main/{locale}/dateFields`
+-   `main/{locale}/numbers`
+-   `main/{locale}/timeZoneNames`
+-   `supplemental/likelySubtags`
+-   `supplemental/numberingSystems`
+-   `supplemental/ordinals`
+-   `supplemental/plurals`
+-   `supplemental/timeData`
+-   `supplemental/weekData`
+
+For [number/currency formatting](#date-and-number-formatting):
+
+-   `main/{locale}/currencies`
+-   `main/{locale}/numbers`
+-   `supplemental/currencyData`
+-   `supplemental/likelySubtags`
+-   `supplemental/numberingSystems`
+-   `supplemental/ordinals`
+-   `supplemental/plurals`
+
+For [unit formatting](#date-and-number-formatting):
+
+-   `main/{locale}/numbers`
+-   `main/{locale}/units`
+-   `supplemental/likelySubtags`
+-   `supplemental/numberingSystems`
+-   `supplemental/ordinals`
+-   `supplemental/plurals`
+
+## Message Formatting
+
+### Basic token replacement
+
+The `i18n` module exposes two methods that handle message formatting: 
+- `formatMessage`, which directly returns a formatted message based on its inputs
+- `getMessageFormatter`, which returns a method dedicated to formatting a single message.
+  
+Both of these methods operate on bundle objects, which must first be registered with the i18n ecosystem by passing them to the `i18n` function (see below).
+
+`@dojo/framework/i18n` supports the [ICU message format](#icu-message-formatting), but that requires CLDR data and is not something that every application requires. As such, if the `supplemental/likeSubtags` and `supplemental/plurals` data are not loaded, then both `formatMessage` and `getMessageFormatter` will perform simple token replacement.
+
+For example, given a `guestInfo` message:
+
+>`{host} invites {guest} to the party.`
+
+an object with `host` and `guest` properties can be provided to a formatter without the need to load CLDR data:
+
+```ts
+import i18n, { formatMessage, getMessageFormatter } from '@dojo/framework/i18n/i18n';
+import bundle from 'nls/main';
+
+i18n(bundle, 'en').then(() => {
+	const formatter = getMessageFormatter(bundle, 'guestInfo', 'en');
+	let message = formatter({
+		host: 'Margaret Mead',
+		guest: 'Laura Nader'
+	});
+	console.log(message); // "Margaret Mead invites Laura Nader to the party."
+
+	// Note that `formatMessage` is essentially a convenience wrapper around `getMessageFormatter`.
+	message = formatMessage(
+		bundle,
+		'guestInfo',
+		{
+			host: 'Marshall Sahlins',
+			gender: 'male',
+			guest: 'Bronisław Malinowski'
+		},
+		'en'
+	);
+	console.log(message); // "Marshall Sahlins invites Bronisław Malinowski to the party."
+});
+```
+
+### ICU Message Formatting
+
+**Note**: This feature requires appropriate [CLDR data](#loading-cldr-data) to have been loaded into the application.
```

Comment:
I presume all the formatters that work on message bundles can be also done using `format` returned from the `this.localizeBundle`? We should mention that if we are going to demonstrate the formatters using `i18n` outside of a widget? Or we should demonstrate them from the context of a widget and mention that they can be used standalone?

Reply:
I've added examples for message formatting via `localizedBundle`.`format` in widgets as the primary mechanism. The existing examples are secondary as 'direct usage'.